From f14b5fba2dd74adcd96ddceff85f9e986c8b0d1f Mon Sep 17 00:00:00 2001
From: Dan Milea <dan.milea@ti.com>
Date: Tue, 1 Mar 2011 11:05:09 -0600
Subject: OMAP4: DSS: Fix fault in display controller plane setup

_dispc_setup_plane: Since the "yuv2rgb_conv" variable is a client provided
pointer, a NULL check should be done before any attempt to de-reference the
pointer.

Change-Id: I456456b233c02591365b877c2d71a85a3c434313
Signed-off-by: Dan Milea <dan.milea@ti.com>
---
 drivers/video/omap2/dss/dispc.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/video/omap2/dss/dispc.c b/drivers/video/omap2/dss/dispc.c
index 5abf4e3..ca1ee1c 100644
--- a/drivers/video/omap2/dss/dispc.c
+++ b/drivers/video/omap2/dss/dispc.c
@@ -3284,7 +3284,7 @@ static int _dispc_setup_plane(enum omap_plane plane,
 				   scale_x, scale_y);
 		_dispc_set_vid_size(plane, out_width, out_height);
 
-		if (yuv2rgb_conv->dirty) {
+		if (yuv2rgb_conv && yuv2rgb_conv->dirty) {
 			struct omap_dss_color_conv cc;
 			int i;
 
-- 
1.6.0.4

